src/rules/expect.ts

```diff
@@ -100,17 +100,18 @@ function validate(context: TSESLint.RuleContext<MessageIds, [Options]>, options:
   }
 
   const checker = program.getTypeChecker();
+  const languageService = ts.createLanguageService(getLanguageServiceHost(program));
   // Don't care about emit errors.
   const diagnostics = ts.getPreEmitDiagnostics(program, sourceFile);
-  if (sourceFile.isDeclarationFile || !/\$Expect(Type|Error)/.test(sourceFile.text)) {
+  if (sourceFile.isDeclarationFile || !/(?:\$Expect(Type|Error|^\?))|\^\?/.test(sourceFile.text)) {
```

Comment:
Existing issue: the lint rule won't run on declaration files! Doesn't seem to me like there's a reason to stop on `sourceFile.isDeclarationFile`. I can file an issue.

Reply:
#48 ✨